pkg/controllers/management/auth/globalroles/enqueue_test.go

```diff
@@ -497,3 +573,336 @@ func Test_crtbEnqueueGRB(t *testing.T) {
+				},
+			},
+			stateSetup: func(state testState) {
+				grb := &v3.GlobalRoleBinding{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-GRB",
+					},
+				}
+				state.grbCacheMock.EXPECT().GetByIndex(grbSafeConcatIndex, "test-GRB").Return([]*v3.GlobalRoleBinding{grb}, nil)
+			},
+			wantKeys:  []relatedresource.Key{{Name: "test-GRB"}},
+			wantError: false,
+		},
+		{
+			name: "GRB with long name gets enqueued",
+			inputObject: &v1.RoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "test-roleBinding",
+					Labels: map[string]string{grbOwnerLabel: safeConcatLongName},
+				},
+			},
+			stateSetup: func(state testState) {
+				grb := &v3.GlobalRoleBinding{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: longName,
+					},
+				}
+				state.grbCacheMock.EXPECT().GetByIndex(grbSafeConcatIndex, safeConcatLongName).Return([]*v3.GlobalRoleBinding{grb}, nil)
+			},
+			wantKeys:  []relatedresource.Key{{Name: longName}},
+			wantError: false,
+		},
+	}
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			ctrl := gomock.NewController(t)
+			grbCache := fake.NewMockNonNamespacedCacheInterface[*v3.GlobalRoleBinding](ctrl)
+			state := testState{
+				grbCacheMock: grbCache,
+			}
+			if test.stateSetup != nil {
+				test.stateSetup(state)
+			}
+			enqueuer := globalRBACEnqueuer{
+				grbCache: grbCache,
+			}
+			res, resErr := enqueuer.roleBindingEnqueueGRB("", "", test.inputObject)
+			require.Len(t, res, len(test.wantKeys))
+			for _, key := range test.wantKeys {
+				require.Contains(t, res, key)
+			}
+			if test.wantError {
+				require.Error(t, resErr)
+			} else {
+				require.NoError(t, resErr)
+			}
+		})
+	}
+}
+
+func Test_namespaceEnqueueGR(t *testing.T) {
+	t.Parallel()
+	type testState struct {
+		grCacheMock *fake.MockNonNamespacedCacheInterface[*v3.GlobalRole]
+	}
+	tests := []struct {
+		name        string
+		stateSetup  func(state testState)
+		inputObject runtime.Object
+		wantKeys    []relatedresource.Key
+		wantError   bool
+	}{
+		{
+			name:        "object is nil",
+			inputObject: nil,
+			wantError:   false,
+		},
+		{
+			name:        "object is not a namespace",
+			inputObject: &v3.GlobalRole{},
+			wantError:   false,
+		},
+		{
+			name: "namespace not in a NamespacedRule",
+			inputObject: &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "ns1",
+				},
+			},
+			stateSetup: func(state testState) {
+				state.grCacheMock.EXPECT().GetByIndex(grNsIndex, "ns1").Return([]*v3.GlobalRole{}, nil)
+			},
+			wantError: false,
+		},
+		{
+			name: "get namespace fails",
+			inputObject: &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "ns1",
+				},
+			},
+			stateSetup: func(state testState) {
+				state.grCacheMock.EXPECT().GetByIndex(grNsIndex, "ns1").Return(nil, fmt.Errorf("error"))
+			},
+			wantError: true,
+		},
+		{
+			name: "multiple grs are returned",
```

Comment:
Is this test case valid? `ns1` was enqueued, but we enqueued two GlobalRoles, neither of which have any rules for `ns1`.

I realize that the indexer, which you are mocking here, is the source of truth for this, but I think that it's a bit odd to have a test case asserting behavior that would be a bug in production.